ambry-account/src/main/java/com/github/ambry/account/MySqlAccountService.java

```diff
@@ -553,31 +955,85 @@ public Container getContainerByName(String accountName, String containerName) th
    */
   @Override
   public Container getContainerById(short accountId, Short containerId) throws AccountServiceException {
-    Account account = getAccountById(accountId);
-    if (account == null) {
-      return null;
-    }
-    Container container = account.getContainerById(containerId);
-    if (container == null) {
-      // If container is not present in the cache, query from mysql db
-      try {
-        container = mySqlAccountStore.getContainerById(accountId, containerId);
-        if (container != null) {
-          // write container to in-memory cache
-          updateContainersInCache(Collections.singletonList(container));
-          logger.info("Container Id {} in Account {} is not found locally, fetched from mysql db", containerId,
-              account.getName());
-          accountServiceMetrics.onDemandContainerFetchCount.inc();
-        } else {
-          logger.error("Container Id {} is not found in Account {}", containerId, account.getName());
-          // Note: We are not using LRU cache for storing recent unsuccessful get attempts by container Ids since this
-          // will be called in getBlob() path which should have valid account-container in most cases.
+    if (config.enableNewDatabaseForMigration) {
+      Account account = getAccountByIdNew(accountId);
+      Container container = null;
+      if (account != null) {
+        container = account.getContainerById(containerId);
+      }
+      if (container == null) {
+        // If container is not present in the cache, query from mysql db
+        try {
+          container = mySqlAccountStoreNew.getContainerById(account.getId(), containerId);
+          if (container != null) {
+            // write container to in-memory cache
+            updateContainersInCacheNew(Collections.singletonList(container));
+            logger.info("Container Id {} in Account {} is not found locally, fetched from new mysql db", containerId,
+                account.getName());
+            accountServiceMetrics.onDemandContainerFetchCountNew.inc();
+          }
+        } catch (SQLException e) {
+          throw translateSQLException(e);
+        }
+        logger.trace("Found container from new db, container: " + containerId);
+        return container;
```

Comment:
Same as above. A helper function can simplify the code and improve readability

Reply:
Add the helper function for old db and later will remove it after migration is fully tested.